### DISCUSSION TITLE
Any updates on the probabilistic modelling tutorials for GemPy?

### DISCUSSION THREAD

**Discussion by Martytalius** [gempy-project/gempy#657] [Q&A]
Hi all, 

Just curious how far along a tutorial for probabilistic modelling has been coming? I would love to get some guidance on implementing probabilistic modelling in my own workflows. 

Thanks again so much to the team and contributors for working on such an amazing opensource modelling software for geology.

**Comment by AlexanderJuestel** [comment]
@flohorovicic @Leguark

**Reply by flohorovicic** [reply]
Thanks for the tag, @AlexanderJuestel - and great to hear your interest in the probabilistic modeling part, @Martytalius! There is currently a major effort in restructuring the GemPy backend by @leguark and it will make sense to wait a bit until these changes are implemented - but we have tutorials with videos (also on the probabilistic methods) very high on our list.

In the meantime, there are examples in the Transform videos - not sure if they still work exactly as shown (as the API since changed a bit), but maybe a good way to get started:

https://youtu.be/n0btC5Zilyc?t=8927

Hope that helps - happy modeling!

**Reply by Martytalius** [reply]
Hey @flohorovicic , thanks for the response! Do you know if there's been any updates in this space coming into 2023?

Love the work of you and your team

**Comment by flohorovicic** [comment]
Dear @Martytalius,

Can't believe it is already one year since the last answer... we are, in fact, still restructuring GemPy for the new version. But for a practical start, maybe @Japhiolite can provide a couple of pointers.

Did you manage to go through the Transform tutorial? I believe that this is partly still based on an older version of PyMC (one of the problems is that, also PyMC, had some major restructuring efforts - which are fantastic, but also had some adjustments in the API, which now results in a loss of full backwards compatibility).

I am also just preparing a couple of notebooks for probabilistic inference with the new(er) PyMC versions. So, aside from all the restructuring of GemPy, we will have more examples. @AlexanderJuestel : maybe also something to document for Jan?

**Reply by AlexanderJuestel** [reply]
@flohorovicic that is something we can certainly think about for @JanWagner1312

**Reply by flohorovicic** [reply]
Also maybe in collaboration with Adam - would fit perfectly as a start into the project.

**Reply by Martytalius** [reply]
Hey @flohorovicic and @Japhiolite , has there been any updates moving into the end of 2023? I am still very keen to use stochastic modelling tutorials as a basis for further use of GemPy :)
